sickbeard/properFinder.py

```diff
@@ -51,86 +53,102 @@ def run(self, force=False):  # pylint: disable=unused-argument
 
         :param force: Start even if already running (currently not used, defaults to False)
         """
-        logger.log(u"Beginning the search for new propers")
+        logger.log('Beginning the search for new propers')
 
         if self.amActive:
-            logger.log(u"Find propers is still running, not starting it again", logger.DEBUG)
+            logger.log('Find propers is still running, not starting it again', logger.DEBUG)
             return
 
         if sickbeard.forcedSearchQueueScheduler.action.is_forced_search_in_progress():
-            logger.log(u"Manual search is running. Can't start Find propers", logger.WARNING)
+            logger.log("Manual search is running. Can't start Find propers", logger.WARNING)
             return
 
         self.amActive = True
 
-        propers = self._getProperList()
+        propers = self._get_proper_results()
 
         if propers:
             self._downloadPropers(propers)
 
         self._set_lastProperSearch(datetime.datetime.today().toordinal())
 
-        run_at = ""
+        run_at = ''
         if None is sickbeard.properFinderScheduler.start_time:
             run_in = sickbeard.properFinderScheduler.lastRun + sickbeard.properFinderScheduler.cycleTime - datetime.datetime.now()
             hours, remainder = divmod(run_in.seconds, 3600)
             minutes, seconds = divmod(remainder, 60)
-            run_at = u", next check in approx. " + (
-                "%dh, %dm" % (hours, minutes) if 0 < hours else "%dm, %ds" % (minutes, seconds))
+            run_at = ', next check in approx. {0}'.format(
+                '{0}h, {1}m'.format(hours, minutes) if 0 < hours else '{0}m, {1}s'.format(minutes, seconds))
 
-        logger.log(u"Completed the search for new propers%s" % run_at)
+        logger.log('Completed the search for new propers{0}'.format(run_at))
 
         self.amActive = False
 
-    def _getProperList(self):  # pylint: disable=too-many-locals, too-many-branches, too-many-statements
+    def _get_proper_results(self):  # pylint: disable=too-many-locals, too-many-branches, too-many-statements # getProperList
         """
-        Walk providers for propers
+        Retrieve a list of recently aired episodes, and search for these episodes in the different providers.
         """
         propers = {}
 
-        search_date = datetime.datetime.today() - datetime.timedelta(days=2)
-
         # for each provider get a list of the
         original_thread_name = threading.currentThread().name
         providers = enabled_providers('backlog')
+
+        # Get the recently aired (last 2 days) shows from db
+        search_date = datetime.datetime.today() - datetime.timedelta(days=2)
+        main_db_con = db.DBConnection()
+        search_qualities = ",".join(set(str(x) for x in Quality.DOWNLOADED + Quality.SNATCHED + Quality.SNATCHED_BEST))
```

Comment:
Moved the query below from GenericProvider to here, because there is no use of doing this query multiple times (for each provider) if the outcome is the same each time. I've made it a set, because there where overlapping status.